### ISSUE TITLE
`manual_slice_fill` false negative

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16924]
### Summary

`clippy::manual_slice_fill` misses simple manual slice fill patterns.

Context: https://github.com/rust-lang/libc/commit/3b53a7b5d588a38716e69d6e9111cda3fc58e4a8
Reproducer: https://rust.godbolt.org/z/4szM9PTnv

### Lint Name

`manual_slice_fill`

### Reproducer

I tried this code:

```rust
pub fn f1(s: &mut [u8; 1]) {
    for slot in s {
        *slot = 0;
    }
}
```

I expected to see this happen:

```rust
pub fn f1(s: &mut [u8; 1]) {
    s.fill(0);
}
```

Instead, this happened:

No lint triggered

### Version

```text
clippy 0.1.97 (52b6e2c208 2026-04-27)
```
